### PR TITLE
[v1.6.x][Bug Fixed] Fix batch norm when grad_req is `add` (#18518)

### PR DESCRIPTION
**Pull request by ChaiBapchya**
* fix batch norm when fix_gamma is True

* support gradient accumulation for batch norm

* mkldnn batchnorm support grad add

* unittest for bn

* fix bn arg

* fix lint

* fix mkldnn

* fix mkldnn bn

* fix grad when fixing gamma

* fix naive gpu bn

* fix lint

* fix cudnn bn

* fix flag

* fix lint

* fix testcase

* fix

* use @pytest.mark.parametrize

* combination

* remove redundant test in batchnorm

* npx.batch_norm test

* try to fix test

* reduce the number of tests for batchnorm

* fix

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by wkcn** [comment]
Hi @ChaiBapchya , the PR #18500 needs to be modified to fit the old version of MXNet.
We can back port the modified PR for MXNet v1.x: https://github.com/apache/incubator-mxnet/pull/18518

**Comment by ChaiBapchya** [comment]
@wkcn cherry-picking 1.x into 1.6 doesn't quite resolve CI issues.. any idea?

**Comment by wkcn** [comment]
Hi @ChaiBapchya ,
I found a compilation error in http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-cpu/detail/PR-18714/3/pipeline
```
[2020-07-18T22:39:10.415Z] src/operator/quantization/mkldnn/../../nn/mkldnn/mkldnn_batch_norm-inl.h: In function 'void mxnet::op::MKLDNNBatchNormBackward(const mxnet::OpContext&, const mxnet::op::BatchNormParam&, const std::vector<mxnet::NDArray>&, const std::vector<mxnet::NDArray>&, const std::vector<mxnet::NDArray>&, const std::vector<mxnet::OpReqType>&, const std::vector<mxnet::NDArray>&, const std::vector<mxnet::NDArray>&)':

[2020-07-18T22:39:10.415Z] src/operator/quantization/mkldnn/../../nn/mkldnn/mkldnn_batch_norm-inl.h:383:36: error: 'copy_size' was not declared in this scope

[2020-07-18T22:39:10.415Z]            memcpy(w_grad_1, gw_buf, copy_size);

[2020-07-18T22:39:10.415Z]                                     ^

[2020-07-18T22:39:10.415Z] src/operator/quantization/mkldnn/../../nn/mkldnn/mkldnn_batch_norm-inl.h:399:46: error: 'copy_size' was not declared in this scope

[2020-07-18T22:39:10.415Z]          memcpy(w_grad_2, &gw_buf[channels_], copy_size);

[2020-07-18T22:39:10.415Z]                                               ^

[2020-07-18T22:39:10.415Z] Makefile:555: recipe for target 'build/src/operator/quantization/mkldnn/mkldnn_quantized_batch_norm.o' failed
```

**Comment by wkcn** [comment]
Hi @ChaiBapchya , all tests passed : )

**Comment by ChaiBapchya** [comment]
@wkcn Appreciate your help with this cherry-pick. Changes look good to me. Can we merge it?

**Comment by wkcn** [comment]
@ChaiBapchya , This PR looks good to me too. : )
